lib/slavery/connection_holder.rb

```diff
@@ -4,19 +4,23 @@ class ConnectionHolder < ActiveRecord::Base
 
     class << self
       # for delayed activation
-      def activate
-        spec = ActiveRecord::Base.configurations[Slavery.spec_key]
+      def activate(target = nil)
+        spec = ActiveRecord::Base.configurations[Slavery.spec_key_for(target)]
         raise Error.new('Slavery.spec_key invalid!') if spec.nil?
         establish_connection spec
       end
     end
   end
 
   class << self
-    def connection_holder
-      @connection_holder ||= begin
-        ConnectionHolder.activate
-        ConnectionHolder
+    def connection_holder(target)
+      slave_connections[target] ||= begin
+        klass = Class.new(Slavery::ConnectionHolder) do
+          self.abstract_class = true
+        end
+        Object.const_set("SlaveryConnection#{target.to_s.camelize}", klass)
```

Comment:
Raising here means we need to clear out all constants created by this block of code in specs. Looking for guidance.